src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs

```diff
@@ -482,206 +421,174 @@ public override Color ForeColor
         ///  current minimum date.
         /// </summary>
-        [
-        SRCategory(nameof(SR.CatBehavior)),
-        SRDescription(nameof(SR.MonthCalendarMaxDateDescr))
-        ]
+        [SRCategory(nameof(SR.CatBehavior))]
+        [SRDescription(nameof(SR.MonthCalendarMaxDateDescr))]
         public DateTime MaxDate
         {
-            get
-            {
-                return DateTimePicker.EffectiveMaxDate(maxDate);
-            }
+            get => DateTimePicker.EffectiveMaxDate(_maxDate);
             set
             {
-                if (value != maxDate)
+                if (value == _maxDate)
                 {
-                    if (value < DateTimePicker.EffectiveMinDate(minDate))
-                    {
-                        throw new ArgumentOutOfRangeException(nameof(value), value, string.Format(SR.InvalidLowBoundArgumentEx, nameof(MaxDate), FormatDate(value), nameof(MinDate)));
-                    }
-                    maxDate = value;
-                    SetRange();
+                    return;
+                }
+
+                if (value < DateTimePicker.EffectiveMinDate(_minDate))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, string.Format(SR.InvalidLowBoundArgumentEx, nameof(MaxDate), FormatDate(value), nameof(MinDate)));
                 }
+
+                _maxDate = value;
+                SetRange();
             }
         }
 
         /// <summary>
         ///  The maximum number of days that can be selected in a
-        ///  month calendar control.  This method does not affect the current
+        ///  month calendar control. This method does not affect the current
         ///  selection range.
         /// </summary>
-        [
-        SRCategory(nameof(SR.CatBehavior)),
-        DefaultValue(DEFAULT_MAX_SELECTION_COUNT),
-        SRDescription(nameof(SR.MonthCalendarMaxSelectionCountDescr))
-        ]
+        [SRCategory(nameof(SR.CatBehavior))]
+        [DefaultValue(DefaultMaxSelectionCount)]
+        [SRDescription(nameof(SR.MonthCalendarMaxSelectionCountDescr))]
         public int MaxSelectionCount
         {
-            get
-            {
-                return maxSelectionCount;
-            }
+            get => _maxSelectionCount;
             set
             {
                 if (value < 1)
                 {
                     throw new ArgumentOutOfRangeException(nameof(value), value, string.Format(SR.InvalidLowBoundArgumentEx, nameof(MaxSelectionCount), value.ToString("D"), 1));
                 }
 
-                if (value != maxSelectionCount)
+                if (value == _maxSelectionCount)
+                {
+                    return;
+                }
+
+                if (IsHandleCreated)
                 {
-                    if (IsHandleCreated)
+                    if (unchecked((int)(long)User32.SendMessageW(this, (User32.WM)ComCtl32.MCM.SETMAXSELCOUNT, (IntPtr)value)) == 0)
```

Comment:
Unrelated to the PR, but I think this can be updated to 
```suggestion
                    if (User32.SendMessageW(this, (User32.WM)ComCtl32.MCM.SETMAXSELCOUNT, (IntPtr)value) == IntPtr.Zero)
```